src/Ng/Jobs/Catalog2PackageFixupJob.cs

```diff
@@ -0,0 +1,160 @@
+
+        public Catalog2PackageFixupJob(ITelemetryService telemetryService, ILoggerFactory loggerFactory)
+            : base(telemetryService, loggerFactory)
+        {
+            ThreadPool.SetMinThreads(MaxDegreeOfParallelism, 4);
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng catalog2packagefixup"
+                   + $"-{Arguments.Source} <catalog>"
+                   + $"-{Arguments.Verify} true/false"
+                   + $"-{Arguments.StorageAccountName} <azure-account>"
+                   + $"-{Arguments.StorageKeyValue} <azure-key> "
+                   + $"-{Arguments.StorageContainer} <azure-container>";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            var source = arguments.GetOrThrow<string>(Arguments.Source);
+            var storageAccount = arguments.GetOrThrow<string>(Arguments.StorageAccountName);
+            var storageKey = arguments.GetOrThrow<string>(Arguments.StorageKeyValue);
+            var storageContainer = arguments.GetOrThrow<string>(Arguments.StorageContainer);
+            var verify = arguments.GetOrDefault(Arguments.Verify, false);
+            var verbose = arguments.GetOrDefault(Arguments.Verbose, false);
+
+            // Prepare the catalog reader.
+            var httpMessageHandlerFactory = CommandHelpers.GetHttpMessageHandlerFactory(TelemetryService, verbose);
+            var collectorHttpClient = new CollectorHttpClient(httpMessageHandlerFactory());
+
+            _catalogIndexReader = new CatalogIndexReader(new Uri(source), collectorHttpClient, TelemetryService);
+
+            // Prepare the Azure Blob Storage client.
+            var credentials = new StorageCredentials(storageAccount, storageKey);
+            var account = new CloudStorageAccount(credentials, useHttps: true);
+
+            var container = account
+                .CreateCloudBlobClient()
+                .GetContainerReference(storageContainer);
+
+            // Prepare the HTTP Client
+            var httpClient = new HttpClient(new WebRequestHandler
+            {
+                AllowPipelining = true
+            });
+
+            httpClient.Timeout = TimeSpan.FromMinutes(10);
+
+            // Prepare the handler that will run on each catalog entry.
+            IPackagesContainerHandler handler;
+            if (verify)
+            {
+                Logger.LogInformation("Validating that all packages have the proper Content MD5 hash...");
+
+                handler = new ValidatePackageHashHandler(
+                    httpClient,
+                    TelemetryService,
+                    LoggerFactory.CreateLogger<ValidatePackageHashHandler>());
+            }
+            else
+            {
+                Logger.LogInformation("Ensuring all packages have a Content MD5 hash...");
+
+                handler = new FixPackageHashHandler(
```

Comment:
>handler [](start = 16, length = 7)

So if we want to clean up cache header or anything else, what would we do? enhance `FixPackageHashHandler` or add a new handler?

Reply:
Either approach would work. I’d prefer a new handler to keep each handler simple.